pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,564 @@
+}
+
+const (
+	authorizedKeysMountPath   = "/var/run/authorized_keys.d"
+	reconcileSSHDFinalizerDNS = "dns.cloudingress.managed.openshift.io"
+)
+
+// Reconcile reads that state of the cluster for a SSHD object and makes changes based on the state read
+// and what is in the SSHD.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling SSHD")
+
+	// Fetch the SSHD instance
+	instance := &cloudingressv1alpha1.SSHD{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// Ensure we have an awsClient instance.
+	if r.awsClient == nil {
+		region, err := utils.GetClusterRegion(r.client)
+		if err != nil {
+			r.SetSSHDStatusError(instance, reqLogger, "Failed to get cluster's AWS region", err)
+			return reconcile.Result{}, err
+		}
+		r.awsClient, err = awsclient.GetAWSClient(r.client, awsclient.NewAwsClientInput{
+			SecretName: config.AWSSecretName,
+			NameSpace:  config.OperatorNamespace,
+			AwsRegion:  region,
+		})
+		if err != nil {
+			r.SetSSHDStatusError(instance, reqLogger, "Failed to create an AWS client", err)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Check for a deletion timestamp.
+	if instance.DeletionTimestamp.IsZero() {
+		// Request object is alive, so ensure it has the DNS finalizer.
+		if !controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) {
+			controllerutil.AddFinalizer(instance, reconcileSSHDFinalizerDNS)
+			if err = r.client.Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+	} else {
+		// Request object is being deleted.
+		if controllerutil.ContainsFinalizer(instance, reconcileSSHDFinalizerDNS) && r.route53 != nil {
+			r.SetSSHDStatus(instance, "Deleting DNS aliases", cloudingressv1alpha1.SSHDStateFinalizing)
+			if err = r.deleteDNSRecords(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Remove the DNS finalizer and update the request object.
+			controllerutil.RemoveFinalizer(instance, reconcileSSHDFinalizerDNS)
+			if err = r.client.Update(context.TODO(), instance); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		// Halt the reconciliation.
+		return reconcile.Result{}, nil
+	}
+
+	// List ConfigMaps with SSH keys
```

Comment:
I'd like to see a comment in this part of the code (dealing with keys) that reminds the reader that changes in the configmaps will be reconciled at least once a minute, so a created or deleted key will have that amount of lag time to be noticed by the deployment. The comment could also remind the reader how that process will work (see my comment about spec equality).

Reply:
Good point.  It's also the case that a _new_ configmap (for a new team) will eventually be detected and an updated deployment created.  I'm not explicitly watching for this event (since there's no ownership linkage to the operator), but just relying on the default reconcile frequency.